### PR TITLE
Add plotting example

### LINKED ISSUE #93: Flow Plot Example in Documentation
Due to new feature from #91 a small section or example for plotting flows in networks would be good to include in the documentation. 
As soon as #90 is through we might think of restructuring the [Plotting Networks](https://pypsa.readthedocs.io/en/latest/plotting.html) section a bit.

### PR DESCRIPTION
**Pull request by energyLS**
Closes #93.

## Changes proposed in this Pull Request
This PR adds a new example of the network plotting functionality to the docs based on a .ipynb.

Note: While creating an html file an error occured, which is not linked to this PR:
`docstring of pypsa.linopt.run_and_read_highs:57: ERROR: Unexpected indentation.`

## Checklist

- [ ] Code changes are sufficiently documented; i.e. new functions contain docstrings and further explanations may be given in `doc`.
- [ ] Unit tests for new features were added (if applicable).
- [ ] Newly introduced dependencies are added to `environment.yaml`, `environment_docs.yaml` and `setup.py` (if applicable).
- [ ] A note for the release notes `doc/release_notes.rst` of the upcoming release is included.
- [ ] I consent to the release of this PR's code under the MIT license.

### CODE REVIEW

**Comment by fneum** [comment]
This is still missing the notebook, is it? You might need to force-add the .ipynb as this file type is in .gitignore

**Comment by energyLS** [comment]
> This is still missing the notebook, is it? You might need to force-add the .ipynb as this file type is in .gitignore
Thanks for the hint. Yes it was missing, since the notebook was not finished yet. Now it's added and ready for review.

**Comment by energyLS** [comment]
@FabianHofmann agreed to review it, but I can't add him to the reviewers list lacking the permission to do so

File: examples/notebooks/flow-plot.ipynb [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    "#### Import and optimize a network"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# n = pypsa.Network('/mnt/c/Users/scl38887/Documents/git/PyPSA/examples/notebooks/flow-plot.ipynb')\n",
```
Comment (FabianHofmann) [inline examples/notebooks/flow-plot.ipynb:43]:
```suggestion
```

File: examples/notebooks/flow-plot.ipynb [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+    "n = pypsa.examples.ac_dc_meshed(from_master=True)\n",
+    "n.lopf()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Get mean generator power by bus and carrier"
```
Comment (FabianHofmann) [inline examples/notebooks/flow-plot.ipynb:52]:
```suggestion
    "Get mean generator power by bus and carrier:"
```

File: examples/notebooks/flow-plot.ipynb [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+   "outputs": [],
+   "source": [
+    "gen = n.generators.assign(g=n.generators_t.p.mean()).groupby([\"bus\", \"carrier\"]).g.sum()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Plot the electricty flows"
```
Comment (FabianHofmann) [inline examples/notebooks/flow-plot.ipynb:68]:
```suggestion
    "Plot the electricity flows:"
```

File: examples/notebooks/flow-plot.ipynb [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    "    link_widths=0,\n",
+    ")\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Plot the electricity flows with different projection and geomap color"
```
Comment (FabianHofmann) [inline examples/notebooks/flow-plot.ipynb:93]:
```suggestion
    "Plot the electricity flows with a different projection and a colored map:"
```

File: examples/notebooks/flow-plot.ipynb [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+    "    color_geomap=True,\n",
+    ")\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Set arbitary values as flow argument using the MultiIndex of n.branches()"
```
Comment (FabianHofmann) [inline examples/notebooks/flow-plot.ipynb:120]:
```suggestion
    "Set arbitrary values as flow argument using the MultiIndex of `n.branches()`:"
```

File: examples/notebooks/flow-plot.ipynb [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+    "    color_geomap=True,\n",
+    ")\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Adjust link colors according to their mean load"
```
Comment (FabianHofmann) [inline examples/notebooks/flow-plot.ipynb:165]:
```suggestion
    "Adjust link colors according to their mean load:"
```

**Review by FabianHofmann — CHANGES_REQUESTED** [review]

**Comment by energyLS** [comment]
@FabianHofmann thanks for the review, all the suggestions are committed.

**Review by FabianHofmann — APPROVED** [review]

**Pull request merged**